### ISSUE TITLE
LibGit2 Clone Fails: "Unable to exchange encryption keys"

### ISSUE THREAD

**Issue by jmulcahy** [JuliaLang/Pkg.jl#3044]
Cloning git repos from a private Gitlab instance via Pkg is failing for me on Julia 1.6 and 1.7. The current Julia 1.8 beta works for some reason. Setting `JULIA_PKG_USE_CLI_GIT=true` with Julia 1.7 also works. I'm using an RSA SSH key. 

The error returned is `GitError(Code:ERROR, Class:SSH, failed to start SSH session: Unable to exchange encryption keys)`. 

I have no doubt this is due to my peculiar and specific Gitlab set up, but I don't know how to debug this. Any ideas?

**Comment by barucden** [comment]
#3030 is this related?

**Comment by jmulcahy** [comment]
I don't think so. It worked on Julia 1.8 on a machine that only had an RSA key, but not 1.6 and 1.7 with that same key. Maybe it works in 1.8 due to some other upgrade that has happened in the SSH stack?

**Comment by judober** [comment]
I have the same Problem since 1.10. Setting `JULIA_PKG_USE_CLI_GIT` helped.
